spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindIncreasedAccessibilityOfMethods.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * SpotBugs - Find bugs in Java programs
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+package edu.umd.cs.findbugs.detect;
+
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.Detector;
+import edu.umd.cs.findbugs.ba.AnalysisContext;
+import edu.umd.cs.findbugs.ba.ClassContext;
+
+import org.apache.bcel.Repository;
+import org.apache.bcel.classfile.JavaClass;
+import org.apache.bcel.classfile.Method;
+
+import java.util.Arrays;
+
+public class FindIncreasedAccessibilityOfMethods implements Detector {
+
+    private final BugReporter bugReporter;
+
+    public FindIncreasedAccessibilityOfMethods(BugReporter bugReporter) {
+        this.bugReporter = bugReporter;
+    }
+
+    @Override
+    public void visitClassContext(ClassContext classContext) {
+        JavaClass directSuperClass;
+        JavaClass subClass;
+        JavaClass[] superClasses;
+        subClass = classContext.getJavaClass();
+        try {
+            directSuperClass = subClass.getSuperClass();
+            superClasses = subClass.getSuperClasses();
+        } catch (ClassNotFoundException e) {
+            AnalysisContext.reportMissingClass(e);
+            return;
+        }
+
+        for (JavaClass javaClass : superClasses) {
+            for (Method superClassMethod : javaClass.getMethods()) {
+                if (isAccessibleFromSubClass(superClassMethod, subClass, directSuperClass)) {
+                    for (Method subClassMethod : subClass.getMethods()) {
+                        if (isOverridable(superClassMethod, subClassMethod) && !isCloneMethodFromCloneable(subClass, superClassMethod)) {
+                            if (superClassMethod.isProtected() && subClassMethod.isPublic()) {
+                                bugReporter.reportBug(new BugInstance(this, "IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY", NORMAL_PRIORITY)
+                                        .addClassAndMethod(subClass, subClassMethod)
+                                        .addString(subClassMethod.getName())
+                                        .addString("protected")
+                                        .addString("public"));
```

Comment:
Honestly, I don't know if this would be useful information. The class and method already precisely define the method which is problematic. Could you please describe a use case when it would be useful, if you insist on it?

Reply:
By adding source line infos you can specify more precisely the location of the bugreport. Adding the method info already adds source line info for the entire method, which is a wider range of sourcelines. You can add the first sourceline, but it can be a bit problematic, when you only have the header and no body for the function. It can be okay like this, if you prefer it this way 😄

Reply:
Thanks for the clarification! I would like to keep it this way then. 😄